### PR TITLE
CLOUDP-76456: Fix testing for cloud manager with remote host

### PR DESCRIPTION
**Pull request by gssbzn**
## Proposed changes

_Jira ticket:_ CLOUDP-76456

## Checklist

- [x] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added any necessary documentation (if appropriate)
- [x] I have run `make fmt` and formatted my code

### CODE REVIEW

File: build/ci/install-agent-spawn-host.sh [hunk 1]
```diff
@@ -41,7 +41,7 @@ EOF
 for host in ${hosts}; do
     echo "Seeding ${host}"
     ./ego seed "${user}@${host}"
-    # start the app server on the remote host. Choose port 9443 as it is open on the security group.
```
Comment (gssbzn) [inline build/ci/install-agent-spawn-host.sh:44]: not true and not making sense here

File: build/ci/ego [hunk 2]
```diff
@@ -268,7 +261,7 @@ resolve_system_dependencies() {
     if [[ "${PLATFORM}" = "debian" ]] || [[ "${PLATFORM}" = "ubuntu" ]]; then # Debian/Ubuntu
         _wait_for_process_to_finish apt
         _wait_for_process_to_finish dpkg
-        apt-get install -y -q --no-install-recommends ca-certificates curl logrotate openssl snmp
```
Comment (gssbzn) [inline build/ci/ego:271]: `ca-certificates` was giving some issues, also curl and logrotate are already installed so I'm cleaning this up

File: build/ci/ego [hunk 3]
```diff
@@ -207,14 +208,6 @@ _determine_architecture() {
         PLATFORM="amazon"
         OSVER=2
 
-    elif [[ -f /etc/os-release ]] && grep PRETTY /etc/os-release | grep -q 'Amazon Linux AMI' > /dev/null; then # Amazon Linux 1
```
Comment (gssbzn) [inline build/ci/ego:210]: amazon1 is deprecated

File: build/ci/ego [hunk 4]
```diff
@@ -135,9 +139,6 @@ _determine_architecture() {
         elif "$LSB_RELEASE" -a 2>&1 | grep 16.04 > /dev/null 2>&1; then
             OSVER=16
 
-        elif "$LSB_RELEASE" -a 2>&1 | grep 14.04 > /dev/null 2>&1; then
```
Comment (gssbzn) [inline build/ci/ego:138]: ubuntu 14 is deprecated

File: build/ci/ego [hunk 5]
```diff
@@ -123,7 +123,11 @@ _determine_architecture() {
 
     elif uname -a | grep -q Ubuntu > /dev/null 2>&1; then # Ubuntu
         PLATFORM="ubuntu"
-        if "$LSB_RELEASE" -a 2>&1 | grep 19.04 > /dev/null 2>&1; then
+
+        if "$LSB_RELEASE" -a 2>&1 | grep 20.04 > /dev/null 2>&1; then
```
Comment (gssbzn) [inline build/ci/ego:127]: adding ubuntu 20

**Review by andreaangiolillo — APPROVED** [review]
LGTM

**Pull request merged**